hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestSplitOrMergeAtTableLevel.java

```diff
@@ -157,15 +157,17 @@ private void trySplitAndEnsureItFails(final TableName tableName) throws Exceptio
     List<RegionInfo> regions = admin.getRegions(tableName);
     int originalCount = regions.size();
 
-    // split the table and make sure region count does not increase
-    Future<?> f = admin.splitRegionAsync(regions.get(0).getEncodedNameAsBytes(), Bytes.toBytes(2));
     try {
+      // split the table and make sure region count does not increase
+      Future<?> f =
```

Comment:
Had to modify tests as in branch-2, the HBTU creates an instance of HBaseAdmin whose methods splitRegionAsync/mergeRegionsAsync seem to propagate error immediately without a call to future.get() which is in contrast to how the admin instance of HBTU works in 3.x.

CC: @Apache9

Reply:
Gentle ping @Apache9 are we good with this?

Reply:
There are some behavior changes when reimplementing sync client on top of async client. I think it is OK.

Reply:
Thank you @Apache9 for the heads up.